opensrp-reveal/src/main/java/org/smartregister/reveal/server/FileHTTPServer.java

```diff
@@ -0,0 +1,157 @@
+import org.smartregister.reveal.application.RevealApplication;
+import org.smartregister.util.Utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/**
+ * A minimal HTTP server that serves a static file (e.g mapbox style json file)
+ */
+public class FileHTTPServer {
+    public static final int PORT = 9783;
+
+    private static final String DEFAULT_STYLE_JSON_FILE = "map-download-style.json";
+    private static final String DEFAULT_DG_ID_PLACEHOLDER = "DIGITAL_GLOBE_ID";
+    private static String styleJson;
+
+    private final FileHTTPServer.ServerThread server;
+    private final ServerSocket socket;
+
+    public FileHTTPServer(String styleJsonFile, String  digitalGlobeIdPlaceHolder) throws IOException {
+        socket = createBoundSocket(PORT);
+        if (socket == null) {
+            throw new IOException("Could not find an available port");
+        }
+        server = new ServerThread(socket);
+
+        digitalGlobeIdPlaceHolder = StringUtils.isNotBlank(digitalGlobeIdPlaceHolder) ? digitalGlobeIdPlaceHolder : DEFAULT_DG_ID_PLACEHOLDER;
+        styleJsonFile = StringUtils.isNotBlank(styleJsonFile) ? styleJsonFile : DEFAULT_STYLE_JSON_FILE;
+
+        styleJson = Utils.readAssetContents(RevealApplication.getInstance().getApplicationContext(), styleJsonFile);
+        styleJson = styleJson.replace(digitalGlobeIdPlaceHolder, BuildConfig.DG_CONNECT_ID);
+    }
+
+    public void start() {
+        server.start();
+    }
+
+    public boolean isStarted() {
+        return server.isAlive();
+    }
+
+    /**
+     * Binds a ServerSocket to given port.
+     */
+    protected static ServerSocket createBoundSocket(int port) throws IOException {
+        try {
+            return new ServerSocket(port);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Timber.e("Port %d not available", port);
+        return null;
+    }
+
+    class ServerThread extends Thread {
+        final ServerSocket socket;
+
+        ServerThread(ServerSocket socket) {
+            this.socket = socket;
+        }
+
+        public void run() {
+            try {
+                socket.setReuseAddress(true);
+                Timber.i("Ready for requests on port %d", socket.getLocalPort());
+                while (!isInterrupted()) {
+                    Socket connection = socket.accept();
+                    Timber.i("Accepted a client connection");
+                    new ResponseThread(connection).start();
+                }
+                Timber.i("Server thread interrupted");
+            } catch (IOException e) {
+                Timber.i("Server thread stopped: %s", e.getMessage());
+            }
+        }
+    }
+
+    class ResponseThread extends Thread {
+        final Socket connection;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)